### PR TITLE
:seedling: add clusterDeleting condition and  cleanup-priority annotation

### PR DESCRIPTION
**Pull request by zhiweiyin318**
## Related issue(s)

Fixes #

### CODE REVIEW

File: utils/constants.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+const (
+	// LabelDeletionPriority is a label for the resources deployed in cluster ns which are waiting to be cleaned up
+	// after cluster is deleted.
+	// The value is an integer value, The larger the value, the later the order of deletion.
+	// The deletion order is :
+	// 1. delete resources without this label firstly.
+	// 2. delete resources with invalid value of this label.
+	// 3. delete resources following the priority value. For example, there are 2 manifestWorks, one value is set 100
+	// and another is set 10, the manifestWorks with 10 will be deleted before the one with 100.
+	LabelDeletionPriority string = "open-cluster-management.io/deletion-priority"
```
Comment (qiujian16) [inline utils/constants.go]: seems to me it should be a field rather than label

Comment (zhiweiyin318) [reply utils/constants.go]: changed to annotation with the name cleanup-priority.

File: cluster/v1/types.go [hunk 2]
```diff
@@ -257,3 +257,16 @@ const (
 	// relating to the ManagedCluster is removed when the ManagedCluster is deleted.
 	ManagedClusterFinalizer = "cluster.open-cluster-management.io/api-resource-cleanup"
 )
+
+const (
+	// ManagedClusterConditionDeleting is a condition which means the cluster is in deletion process.
+	ManagedClusterConditionDeleting string = "ManagedClusterIsDeleting"
```
Comment (qiujian16) [inline cluster/v1/types.go]: Deleting is good enough, it is ManagedCluster resource so it is not needed to specify it in the type

File: utils/constants.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+const (
+	// CleanupPriorityAnnotationKey is a annotation for the resources deployed in cluster ns which are waiting to
+	// be cleaned up after cluster is deleted.
+	// The value is an integer value [0,100], The larger the value, the later the order of deletion.
+	// The deletion order is :
+	// 1. delete resources without this annotation firstly.
+	// 2. delete resources with invalid value of this annotation (!= [0,100]).
+	// 3. delete resources following the priority value. For example, there are 2 manifestWorks, one value is set 100
+	// and another is set 10, the manifestWorks with 10 will be deleted before the one with 100.
+	CleanupPriorityAnnotationKey string = "open-cluster-management.io/cleanup-priority"
```
Comment (qiujian16) [inline utils/constants.go]: this should also be in the types, but not here.

File: cluster/v1/types.go (hunk 2 above)
Comment (zhiweiyin318) [reply cluster/v1/types.go]: done

File: utils/constants.go (hunk 3 above)
Comment (zhiweiyin318) [reply utils/constants.go]: moved to types of clusters

**Pull request merged**